src/test/java/com/google/step/youtube/YouTubeClientBuilder.java

```diff
@@ -0,0 +1,146 @@
+package com.google.step.youtube;
+
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.api.client.util.DateTime;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.YouTube.Channels;
+import com.google.api.services.youtube.YouTube.PlaylistItems;
+import com.google.api.services.youtube.YouTube.Search;
+import com.google.api.services.youtube.YouTube.Videos;
+import com.google.api.services.youtube.model.Channel;
+import com.google.api.services.youtube.model.ChannelContentDetails;
+import com.google.api.services.youtube.model.ChannelContentDetails.RelatedPlaylists;
+import com.google.api.services.youtube.model.ChannelListResponse;
+import com.google.api.services.youtube.model.PlaylistItem;
+import com.google.api.services.youtube.model.PlaylistItemListResponse;
+import com.google.api.services.youtube.model.PlaylistItemSnippet;
+import com.google.api.services.youtube.model.ResourceId;
+import com.google.api.services.youtube.model.SearchListResponse;
+import com.google.api.services.youtube.model.SearchResult;
+import com.google.api.services.youtube.model.Video;
+import com.google.api.services.youtube.model.VideoListResponse;
+import com.google.api.services.youtube.model.VideoSnippet;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Date;
+
+public class YouTubeClientBuilder {
```

Comment:
A builder is a very specific design pattern, which this is not.  (A builder is a mutable object that you can build up something, and then builds an immutable object.)

For classes that are stateless and only contain static methods like this one  are called utility methods... 

Usually if it's a bunch of factory methods, the convention is to pluralize it.  E.g.:

https://guava.dev/releases/19.0/api/docs/com/google/common/base/Suppliers.html

It's named "Suppliers" and has methods that create "Supplier".

In this case your thing is creating objects of different types so maybe "YouTubeClientMocks" would be another a good name.

Alternatively you coudl call it:  YouTubeClientTestingUtil if it contains a grab bag of stuff.

Reply:
Changed to YouTubeClientMocks